pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -178,6 +193,45 @@ func (m Manager) RunFunction(ctx context.Context, req *fnv1beta1.RunFunctionRequ
 	return sr.GetResponse()
 }
 
+func (m *Manager) extractSteps(service any) []any {
```

Comment:
I fell like we overcomplicated things here, the generics should have saved us from reflection....

Reply:
As discussed, generics can't help in this situation unfortunately...